ampd/src/metrics.rs

```diff
@@ -0,0 +1,108 @@
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+
+use lazy_static::lazy_static;
+use prometheus::{
+    self, register_histogram, register_int_counter, register_int_gauge, Encoder, Histogram,
+    HistogramOpts, IntCounter, IntGauge, TextEncoder,
+};
+use tokio::task::JoinHandle;
+use warp::Filter;
+
+lazy_static! {
+    // times it calls bond_verifiers
+    pub static ref VERIFIERS_BONDED: IntCounter = register_int_counter!(
+        "verifiers_bonded_calls_total",
+        "Total number times calling bond_verifiers command"
+    ).unwrap();
+
+     // times it calls verifier_address
+    pub static ref VERIFIER_ADDRESS_CALLS: IntCounter = register_int_counter!(
+        "verifier_address_calls_total",
+        "Total number of times calling the verifier_address command"
+    ).unwrap();
+
+    // time it takes to load configuartion file
+    pub static ref CONFIG_LOAD_TIME: Histogram = register_histogram!(
+        HistogramOpts::new(
+            "config_load_seconds",
+            "Time taken to load and parse configuration files"
+        )
+        .buckets(vec![0.005, 0.01, 0.05, 0.1, 0.5, 1.0])
+    ).unwrap();
+    // 2-secnd updates
+    pub static ref UPTIME_SECONDS: IntGauge = register_int_gauge!(
+        "ampd_uptime_seconds",
+        "Number of seconds the application has been running"
+    ).unwrap();
+}
+
+lazy_static! {
+    static ref START_TIME: Instant = Instant::now();
+}
+
+fn update_metrics() {
+    let uptime = START_TIME.elapsed().as_secs() as i64;
+    UPTIME_SECONDS.set(uptime);
+}
+
+fn start_metrics_updater() {
+    tokio::spawn(async {
+        let mut interval = tokio::time::interval(Duration::from_secs(2));
+
+        loop {
+            interval.tick().await;
+            update_metrics();
+        }
+    });
+}
+
+pub fn increment_verifiers_bonded() {
+    VERIFIERS_BONDED.inc();
+}
+
+pub fn increment_verifier_address_calls() {
+    VERIFIER_ADDRESS_CALLS.inc();
+}
+
+pub struct Timer {
+    start: Instant,
+}
+
+impl Timer {
+    pub fn new() -> Self {
+        Timer {
+            start: Instant::now(),
+        }
+    }
+
+    pub fn elapsed_seconds(&self) -> f64 {
+        self.start.elapsed().as_secs_f64()
+    }
+}
+
+pub fn observe_config_load_time(duration_seconds: f64) {
+    CONFIG_LOAD_TIME.observe(duration_seconds);
+}
+
+pub fn gather_metrics() -> String {
+    let encoder = TextEncoder::new();
+    let mut buffer = Vec::new();
+    let metric_families = prometheus::gather();
+    encoder.encode(&metric_families, &mut buffer).unwrap();
+    String::from_utf8(buffer).unwrap()
+}
+
+pub async fn start_metrics_server(address: Option<SocketAddr>) -> JoinHandle<()> {
+    let addr = address.unwrap_or_else(|| std::net::SocketAddr::from(([127u8, 0, 0, 1], 9898)));
```

Comment:
Should use the config to provide default instead of this kind of hardcoding. It also makes not sense to use `unwrap_or_else` but `wrap_or` in this situation.